src/actions.rs

```diff
@@ -0,0 +1,886 @@
+//! Actions functionality for WebDriver.
+use crate::elements::Element;
+use std::fmt::Debug;
+use std::time::Duration;
+use webdriver::actions as WDActions;
+
+/// An action not associated with an input device (i.e. pause).
+#[derive(Debug, Clone)]
+pub enum NullAction {
+    /// Pause for the specified duration.
+    Pause {
+        /// The pause duration, given in milliseconds.
+        duration: u64,
+    },
+}
+
+impl From<NullAction> for WDActions::NullActionItem {
+    fn from(na: NullAction) -> Self {
+        match na {
+            NullAction::Pause { duration } => WDActions::NullActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration),
+                }),
+            ),
+        }
+    }
+}
+
+/// An action performed with a keyboard.
+#[derive(Debug, Clone)]
+pub enum KeyAction {
+    /// Pause action.
+    /// Useful for adding pauses between other key actions.
+    Pause {
+        /// The pause duration, given in milliseconds.
+        duration: u64,
+    },
+    /// Key up action.
+    Up {
+        /// The key code, e.g. `'a'`. See the [`Keys`] module for special key codes.
+        ///
+        /// [`Keys`]: crate::keys::Keys
+        value: char,
+    },
+    /// Key down action.
+    Down {
+        /// The key code, e.g. `'a'`. See the [`Keys`] module for special key codes.
+        ///
+        /// [`Keys`]: crate::keys::Keys
+        value: char,
+    },
+}
+
+impl From<KeyAction> for WDActions::KeyActionItem {
+    fn from(ka: KeyAction) -> Self {
+        use webdriver::actions::{KeyAction as WDKeyAction, KeyDownAction, KeyUpAction};
+        match ka {
+            KeyAction::Pause { duration } => WDActions::KeyActionItem::General(
+                WDActions::GeneralAction::Pause(WDActions::PauseAction {
+                    duration: Some(duration),
+                }),
+            ),
+            KeyAction::Up { value } => {
+                WDActions::KeyActionItem::Key(WDKeyAction::Up(KeyUpAction {
+                    value: value.to_string(),
+                }))
+            }
+            KeyAction::Down { value } => {
+                WDActions::KeyActionItem::Key(WDKeyAction::Down(KeyDownAction {
+                    value: value.to_string(),
+                }))
+            }
+        }
+    }
+}
+
+/// An action performed with a pointer device. See `PointerActionType` for
+/// the availabledevice types.
+#[derive(Debug, Clone)]
+pub enum PointerAction {
+    /// Pause action.
+    /// Useful for adding pauses between other key actions.
+    Pause {
+        /// The pause duration, given in milliseconds.
+        duration: u64,
+    },
+    /// Pointer button down.
+    Down {
+        /// The mouse button index. Button values are as follows:
```

Comment:
Can this be an `enum`?

Reply:
Hmm, not really. We could provide an enum for it, but AFAIK there's no actual limit to the number of buttons a mouse could have. Also I'm not sure the same enum would even make sense for other pointer types.

An enum with just `Left`, `Middle`, `Right` might be useful, but I think it should still store the button as `u64`.

Reply:
How about including
```rust
pub type MouseButtonLeft = 0;
pub type MouseButtonMiddle = 1;
pub type MouseButtonRight = 2;
```
?

Reply:
I'm ok with this. I'd also be ok with an enum with `#[repr(u64)]` but it's slightly nicer to not have to type `as u64` every time. Let's go with the above for now.

Reply:
Do you mean `pub const` ? This is fine. An enum with `#[repr(u64)]` would also be fine, but slightly annoying to require `as u64` each time.

As a `const` should it be `MOUSE_BUTTON_LEFT` etc?

Reply:
Wow, I must have been low on sleep. Yes, I meant `const`, in which case it should be `MOUSTE_BUTTON_LEFT`. I thought about a `#[repr(u64)]` enum, but I think that'll be annoying enough not to be worthwhile. And you already did it, so :+1: